src/ng-xform/field-components/base-dynamic-field.component.ts

```diff
@@ -1,35 +1,42 @@
 import { Input, OnInit } from '@angular/core';
-import { FormGroup, FormControl } from '@angular/forms';
+import { FormGroup, FormControl, Validators } from '@angular/forms';
 import { DynamicField } from '../fields';
 
 /**
  * Base class for Dynamic for Fields
  */
 export class BaseDynamicFieldComponent<T extends DynamicField> implements OnInit {
-    @Input() field: T;
-    @Input() form: FormGroup;
-    @Input() editing = true;
+  @Input() field: T;
+  @Input() form: FormGroup;
+  @Input() editing = true;
 
-    control: FormControl;
-
-    ngOnInit() {
-      this.control = <FormControl> this.form.controls[this.field.key];
-    }
+  control: FormControl;
 
+  ngOnInit() {
+    this.control = <FormControl>this.form.controls[this.field.key];
+  }
 
-    /**
-     * Property to be used as the Form Element ID
-     */
-    get elementId(): string {
-      return this.field.key;
-    }
+  /**
+   * Property to be used as the Form Element ID
+   */
+  get elementId(): string {
+    return this.field.key;
+  }
 
+  get isOptional(): boolean {
+    return this.editing && !this.field.readOnly &&
```

Comment:
You would avoid testing if it is 'editing' if you remove the 'optional' part from the div in the non-editing state.

Reply:
you're right